Modules/_sre.c

```diff
@@ -1414,12 +1414,16 @@ _sre_compile_impl(PyObject *module, PyObject *pattern, int flags,
         self->code[i] = (SRE_CODE) value;
         if ((unsigned long) self->code[i] != value) {
             PyErr_SetString(PyExc_OverflowError,
-                            "regular expression code size limit exceeded");
+                            "regular expression code out of range");
```

Comment:
One option is to add `Py_DECREF(self);` and `return NULL;` here.

Other option is to left just `break` and rewrite the condition after the loop:
```
if (i < n || PyErr_Occurred() && PyErr_ExceptionMatches(PyExc_OverflowError)) {
    PyErr_SetString(PyExc_OverflowError, ...);
}
if (PyErr_Occurred()) {
    Py_DECREF(self);
    return NULL;
}
```
Third option is to add a label before `PyErr_SetString(PyExc_OverflowError, ...);` and jump to it from here. This looks the simplest option.